iceoryx_posh/include/iceoryx_posh/popo/client.hpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_POPO_CLIENT_HPP
+#define IOX_POSH_POPO_CLIENT_HPP
+
+#include "iceoryx_posh/capro/service_description.hpp"
+#include "iceoryx_posh/internal/popo/client_impl.hpp"
+#include "iceoryx_posh/popo/client_options.hpp"
+#include "iceoryx_posh/popo/trigger_handle.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+
+namespace iox
+{
+namespace popo
+{
+template <typename Req, typename Res>
+class Client : public ClientImpl<Req, Res>
+{
+  public:
+    using ClientImpl<Req, Res>::ClientImpl;
```

Comment:
Why using a class when you could do it like:
```cpp
template<typename Req, typename Res>
using Client = ClientImpl<Req, Res>;
```

Reply:
Good question. Initially there was more code. The Publisher and Subscriber do it this way ... we also had some issues with doxygen not showing the documentation of the alias but I guess that can be solved